ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py

```diff
@@ -606,3 +606,55 @@ def test_paddle_matrix_power(
         x=x[0],
         n=n,
     )
+
+
+# cond
+@st.composite
+def _get_dtype_and_matrix_non_singular(draw, dtypes):
+    while True:
+        matrix = draw(
+            helpers.dtype_and_values(
+                available_dtypes=dtypes,
+                min_value=-10,
+                max_value=10,
+                min_num_dims=2,
+                max_num_dims=2,
+                min_dim_size=1,
+                max_dim_size=5,
+                shape=st.tuples(st.integers(1, 5), st.integers(1, 5)).filter(
+                    lambda x: x[0] == x[1]
+                ),
+                allow_inf=False,
+                allow_nan=False,
+            )
+        )
+        if np.linalg.det(matrix[1][0]) != 0 and matrix_is_stable(matrix[1][0]):
+            break
+
+    return matrix[0], matrix[1]
+
+
+@handle_frontend_test(
+    fn_tree="paddle.tensor.linalg.cond",
+    dtype_and_x=_get_dtype_and_matrix_non_singular(dtypes=["float32", "float64"]),
+    p=st.sampled_from([None, "fro", "nuc", np.inf, -np.inf, 1, -1, 2, -2]),
+    test_with_out=st.just(False),
+)
+def test_paddle_cond(*, dtype_and_x, p, on_device, fn_tree, frontend, test_flags):
+    dtype, x = dtype_and_x
+
+    if not matrix_is_stable(x[0]):
```

Comment:
can we use assume here to filter stable martrix😂 just a final update I guess

Reply:
> can we use assume here to filter stable martrix😂 just a final update I guess

as a decorator?